braindecode/preprocessing/eegprep_preprocess.py

```diff
@@ -0,0 +1,1139 @@
+    no_overwrite : bool
+        If True, will not overwrite an existing entry with the same key.
+
+    """
+    # the description is apparently the only viable place where custom metadata may be
+    # stored in MNE Raw objects
+    description = raw.info.get("description") or ""
+
+    # Try to find existing eegprep data
+    if match := _MARKER_PATTERN.search(description):
+        # Parse existing data
+        try:
+            decoded = base64.b64decode(match.group(1)).decode("utf-8")
+            existing_data = json.loads(decoded, object_hook=_numpy_empty_decoder)
+        except (ValueError, json.JSONDecodeError):
+            existing_data = {}
+        # Check no_overwrite condition
+        if no_overwrite and key in existing_data:
+            return
+        # Update data
+        existing_data[key] = payload
+        new_marker = _encode_payload(existing_data)
+        # Replace the old marker with updated one
+        new_description = _MARKER_PATTERN.sub(new_marker, description, count=1)
+    else:
+        # No existing data, append new marker
+        data = {key: payload}
+        new_marker = _encode_payload(data)
+        # Append with spacing if description exists
+        if description.strip():
+            new_description = f"{description.rstrip()}\n{new_marker}"
+        else:
+            new_description = new_marker
+
+    raw.info["description"] = new_description
+
+
+def _mne_retrieve_json(raw: BaseRaw, *, key: str, delete: bool = False) -> Any | None:
+    """Retrieves data that was previously stored using _mne_stash_json from an MNE
+    BaseRaw dataset.
+
+    Parameters
+    ----------
+    raw : BaseRaw
+        The MNE Raw object to retrieve data from.
+    key : str
+        The key under which the payload was stored.
+    delete : bool
+        If True, removes the key from the stored data after retrieval.
+
+    Returns
+    -------
+    Any | None
+        The retrieved payload, or None if not found.
+    """
+    description = raw.info.get("description") or ""
+    match = _MARKER_PATTERN.search(description)
+    if not match:
+        return None
+
+    try:
+        decoded = base64.b64decode(match.group(1)).decode("utf-8")
+        data = json.loads(decoded, object_hook=_numpy_empty_decoder)
+    except (ValueError, json.JSONDecodeError):
+        return None
+
+    result = data.get(key)
+
+    if delete and key in data:
+        # Remove the key from data
+        del data[key]
+        if data:
+            # Still have other keys, update the marker
+            new_marker = _encode_payload(data)
+            new_description = _MARKER_PATTERN.sub(new_marker, description, count=1)
+        else:
+            # No more keys, remove the entire marker
+            new_description = _MARKER_PATTERN.sub("", description, count=1).strip()
+        raw.info["description"] = new_description
+
+    return result
+
+
+class EEGPrepBasePreprocessor(Preprocessor):
```

Comment:
Here, I think we use a little more of the native function from mne raw to apply, select channels, and reduce by half the amount of code.